pkg/controller/ps/user.go

```diff
@@ -14,13 +17,52 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
 	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/haproxy"
 	"github.com/percona/percona-server-mysql-operator/pkg/k8s"
 	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
 	"github.com/percona/percona-server-mysql-operator/pkg/orchestrator"
+	"github.com/percona/percona-server-mysql-operator/pkg/router"
 	"github.com/percona/percona-server-mysql-operator/pkg/secret"
 	"github.com/percona/percona-server-mysql-operator/pkg/users"
 )
 
+const (
+	annotationPasswordsUpdated string = "percona.com/passwords-updated"
+)
```

Comment:
could you please explain how we use this annotation?

Reply:
This annotation is used if something during pass propagation checks or discarding old passwords, when we come back again (new reconcile loop), we don't repeat updating passwords, updating secrets, and (most importantly) restart replication in case of async replication. I find that it wouldn't be good for the user to repeatedly restart replication in case, say we can't check if the pass is propagated for some time. This is the mechanism to prevent that.

When we see that passwords are propagated and discarded we set this to true. If in this step something fails, we will go back (new reconcile loop) and check this annotation at the beginning again (line 140). So if secret and internal secret are the same and this annotation is false, that basically means that we updated passwords, restarted replication if necessary, and updated internal secrets, but failed to check propagation or discard old passwords, so we just perform these two operations.

Reply:
should we delete the annotation after all work is done? can it interfere with future password updates?

Reply:
As soon as we update passwords we set it to false. We can remove it but the effect is the same and I find it somewhat more clear if we set it to false.